### PR TITLE
[MI-3264] Update webapp base setup to use RTK query

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary

- Updated webapp base setup to use RTK query.
- Added RHS for the plugin.
- Updated `enforceConnectedAccountModal.tsx` component with an updated setup.

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -49,10 +50,12 @@
     "webpack-cli": "5.0.1"
   },
   "dependencies": {
+    "@reduxjs/toolkit": "1.8.2",
     "core-js": "3.29.1",
+    "js-cookie": "3.0.1",
     "mattermost-redux": "5.33.1",
     "react": "18.2.0",
-    "react-redux": "8.0.5",
+    "react-redux": "7.2.0",
```
Comment (raghavaggarwal2308) [inline webapp/package.json:58]: Any reason for downgrading this?

File: webapp/src/hooks/usePluginApi.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import {useSelector, useDispatch} from 'react-redux';
+
+import {setApiRequestCompletionState} from '../reducers/apiRequest';
+
```
Comment (raghavaggarwal2308) [inline webapp/src/hooks/usePluginApi.ts:4]:
```suggestion
```

File: webapp/src/index.tsx [hunk 3]
```diff
@@ -1,26 +1,44 @@
+import React from 'react';
 import {Store, Action} from 'redux';
 
 import {GlobalState} from 'mattermost-redux/types/store';
 
+import Rhs from './containers/Rhs';
+
+import Constants from './constants';
+
+import reducer from './reducers';
```
Comment (raghavaggarwal2308) [inline webapp/src/index.tsx:10]:
```suggestion
import Rhs from './containers/Rhs';
import Constants from './constants';
import reducer from './reducers';
```

File: webapp/src/types/common/service.d.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+type HttpMethod = 'GET' | 'POST' | 'PATCH' | 'DELETE';
+
+type PluginApiServiceName =
+    'needsConnect' |
+    'connect';
```
Comment (raghavaggarwal2308) [inline webapp/src/types/common/service.d.ts:5]: can we make this in one line

File: webapp/src/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import {id as pluginId} from '../manifest';
+
+const getBaseUrls = (): {pluginApiBaseUrl: string} => {
+    const url = new URL(window.location.href);
```
Comment (raghavaggarwal2308) [inline webapp/src/utils/index.ts:4]: We can use the URL from redux here. once [this](https://github.com/Brightscout/mattermost-plugin-template/pull/9) PR is merged. Please add a TODO comment for the same

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/services/index.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import Constants from '../constants';
+import utils from '../utils';
+
+// Service to make plugin API requests
+export const msTeamsPluginApi = createApi({
+    reducerPath: 'msTeamsPluginApi',
+    baseQuery: fetchBaseQuery({
+        baseUrl: utils.getBaseUrls().pluginApiBaseUrl,
+        prepareHeaders: (headers) => {
+            headers.set('X-CSRF-Token', Cookies.get('MMCSRF') ?? '');
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/services/index.ts:14]: This will be used only here. I don't think we need constant for this.

File: webapp/package.json (hunk 1 above)
Comment (ayusht2810) [reply webapp/package.json:58]: we were getting a TypeScript error while dispatching an action in the base setup, which was fixed by downgrading the dependancy.

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by ayusht2810** [comment]
duplicate of PR https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/228. Closing the PR.

**Pull request closed without merging**